File: javaSample/src/main/java/com/cleveradssolutions/sampleapp/appopen/SplashAppOpenAdActivity.java
package com.cleveradssolutions.sampleapp.appopen;

import static com.cleveradssolutions.sampleapp.SampleApplication.CAS_ID;
import static com.cleveradssolutions.sampleapp.SampleApplication.TAG;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.util.Log;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.cleveradssolutions.sampleapp.R;
import com.cleveradssolutions.sampleapp.selection.SelectionActivity;
import com.cleveradssolutions.sdk.AdContentInfo;
import com.cleveradssolutions.sdk.AdFormat;
import com.cleveradssolutions.sdk.screen.CASAppOpen;
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback;
import com.cleversolutions.ads.AdError;

import java.util.concurrent.TimeUnit;

public class SplashAppOpenAdActivity extends Activity {

    private boolean isLoadingAppResources = false;
    private boolean isVisibleAppOpenAd = false;
    private boolean isCompletedSplash = false;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_app_open_ad_activity);

        simulationLongAppResourcesLoading();

        createAppOpenAd();
    }

    @SuppressLint("SetTextI18n")
    private void createAppOpenAd() {
        // Create an Ad
        CASAppOpen appOpenAd = new CASAppOpen(this, CAS_ID);
        appOpenAd.setAutoloadEnabled(false);
        appOpenAd.setAutoshowEnabled(false);

        // Handle fullscreen callback events
        appOpenAd.setContentCallback(new ScreenAdContentCallback() {
            @Override
            public void onAdLoaded(@NonNull AdContentInfo adContentInfo) {
                Log.d(TAG, "App Open Ad loaded");
                if (isLoadingAppResources) {
                    isVisibleAppOpenAd = true;
                    appOpenAd.show(SplashAppOpenAdActivity.this);
                }
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdFormat adFormat, @NonNull AdError adError) {
                Log.e(TAG, "App Open Ad failed to load: " + adError.getMessage());
                startNextActivity();
            }

            @Override
            public void onAdFailedToShow(@NonNull AdFormat adFormat, @NonNull AdError adError) {
                Log.d(TAG, "App Open Ad show failed: " + adError.getMessage());

                isVisibleAppOpenAd = false;
                startNextActivity();
            }

            @Override
            public void onAdShowed(@NonNull AdContentInfo adContentInfo) {
                Log.d(TAG, "App Open Ad shown");
            }

            @Override
            public void onAdClicked(@NonNull AdContentInfo adContentInfo) {
                Log.d(TAG, "App Open Ad clicked");
            }

            @Override
            public void onAdDismissed(@NonNull AdContentInfo adContentInfo) {
                Log.d(TAG, "App Open Ad closed");

                isVisibleAppOpenAd = false;
                startNextActivity();
            }
        });

        appOpenAd.setOnImpressionListener(adContentInfo ->
                Log.d(TAG, "App Open Ad impression: ${ad.revenue}"));

        // Load the Ad
        appOpenAd.load(this);
    }

    private void startNextActivity() {
        if (isLoadingAppResources || isVisibleAppOpenAd || isCompletedSplash)
            return;
        isCompletedSplash = true;
        Intent intent = new Intent(this, SelectionActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
    }

    private void simulationLongAppResourcesLoading() {
        // Simulation of long application resources loading for 5 seconds.
        isLoadingAppResources = true;
        TextView timerText = findViewById(R.id.timerView);
        CountDownTimer timer = new CountDownTimer(TimeUnit.SECONDS.toMillis(5), 1000) {
            @SuppressLint("SetTextI18n")
            @Override
            public void onTick(long millisUntilFinished) {
                timerText.setText(TimeUnit.MILLISECONDS.toSeconds(millisUntilFinished) + " seconds left");
            }

            @Override
            public void onFinish() {
                isLoadingAppResources = false;
                startNextActivity();
            }
        };
        timer.start();

        findViewById(R.id.skipAppOpenAd).setOnClickListener(v -> {
            isLoadingAppResources = false;
            startNextActivity();
        });
    }

}

File: kotlinSample/src/main/java/com/cleveradssolutions/sampleapp/SplashAppOpenAdActivity.kt
package com.cleveradssolutions.sampleapp

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.CountDownTimer
import android.util.Log
import android.view.View
import android.widget.TextView
import com.cleveradssolutions.sampleapp.SampleApplication.Companion.CAS_ID
import com.cleveradssolutions.sampleapp.SampleApplication.Companion.TAG
import com.cleveradssolutions.sampleapp.selection.SelectionActivity
import com.cleveradssolutions.sdk.AdContentInfo
import com.cleveradssolutions.sdk.AdFormat
import com.cleveradssolutions.sdk.OnAdImpressionListener
import com.cleveradssolutions.sdk.screen.CASAppOpen
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback
import com.cleversolutions.ads.AdError
import java.util.concurrent.TimeUnit

class SplashAppOpenAdActivity : Activity() {
    private var isLoadingAppResources = false
    private var isVisibleAppOpenAd = false
    private var isCompletedSplash = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Add <activity ... android:configChanges="orientation ..." />
        // in manifest to avoid calling onCreate() multiple times
        setContentView(R.layout.splash_app_open_ad_activity)

        simulationLongAppResourcesLoading()
        createAppOpenAd()
    }

    @SuppressLint("SetTextI18n")
    private fun createAppOpenAd() {
        // Create an Ad
        val appOpenAd = CASAppOpen(CAS_ID)

        // Handle fullscreen callback events
        appOpenAd.contentCallback = object : ScreenAdContentCallback() {
            override fun onAdLoaded(ad: AdContentInfo) {
                Log.d(TAG, "App Open Ad loaded")
                if (isLoadingAppResources) {
                    isVisibleAppOpenAd = true
                    appOpenAd.show(this@SplashAppOpenAdActivity)
                }
            }

            override fun onAdFailedToLoad(format: AdFormat, error: AdError) {
                Log.e(TAG, "App Open Ad failed to load: ${error.message}")
                startNextActivity()
            }

            override fun onAdFailedToShow(format: AdFormat, error: AdError) {
                Log.e(TAG, "App Open Ad show failed: ${error.message}")
                isVisibleAppOpenAd = false
                startNextActivity()
            }

            override fun onAdShowed(ad: AdContentInfo) {
                Log.d(TAG, "App Open Ad shown")
            }

            override fun onAdClicked(ad: AdContentInfo) {
                Log.d(TAG, "App Open Ad clicked")
            }

            override fun onAdDismissed(ad: AdContentInfo) {
                Log.d(TAG, "App Open Ad closed")
                isVisibleAppOpenAd = false
                startNextActivity()
            }
        }

        appOpenAd.onImpressionListener = OnAdImpressionListener { ad ->
            Log.d(TAG, "App Open Ad impression: ${ad.revenue}")
        }

        appOpenAd.load(this)
    }

    private fun startNextActivity() {
        if (isLoadingAppResources || isVisibleAppOpenAd || isCompletedSplash)
            return
        isCompletedSplash = true
        val intent = Intent(this, SelectionActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        startActivity(intent)
    }

    private fun simulationLongAppResourcesLoading() {
        // Simulation of long application resources loading for 5 seconds.
        isLoadingAppResources = true
        val timerText = findViewById<TextView>(R.id.timerView)
        val timer = object : CountDownTimer(TimeUnit.SECONDS.toMillis(7), 1000) {
            @SuppressLint("SetTextI18n")
            override fun onTick(millisUntilFinished: Long) {
                timerText.text =
                    "${TimeUnit.MILLISECONDS.toSeconds(millisUntilFinished)} seconds left"
            }

            override fun onFinish() {
                isLoadingAppResources = false
                startNextActivity()
            }
        }
        timer.start()

        findViewById<View>(R.id.skipAppOpenAd).setOnClickListener { v: View? ->
            isLoadingAppResources = false
            startNextActivity()
        }
    }
}